profiles/common/flake.nix

```diff
@@ -0,0 +1,66 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.srvos;
+in
+{
+  options.srvos = {
+    flake = lib.mkOption {
+      # FIXME what is the type of a flake?
+      type = lib.types.nullOr lib.types.raw;
+      default = null;
+      description = lib.mdDoc ''
+        Flake that the nixos contains the nixos configuration.
+      '';
+    };
```

Comment:
What do you think of relying on `_module.args.inputs` to exist as a convention instead?

Reply:
Is there a way to detect if `_module.args.inputs` exists? otherwise I am unsure how to make this the default without breaking evaluation

Reply:
with `_module.args ? inputs`, no?

Reply:
but I agree that the relationship is more implicit so it would have to be documented properly

Reply:
We also need `_module.args ? self`. I can make the option default to set...

Reply:
It does not seem to work. I cannot access _module.args.